src/Compilers/CSharp/Portable/Syntax/LookupPosition.cs

```diff
@@ -320,7 +320,7 @@ private static SyntaxToken GetFirstIncludedToken(StatementSyntax statement, bool
                 case SyntaxKind.ReturnStatement:
                     return ((ReturnStatementSyntax)statement).ReturnKeyword;
                 case SyntaxKind.SwitchStatement:
-                    return ((SwitchStatementSyntax)statement).OpenBraceToken;
+                    return ((SwitchStatementSyntax)statement).Expression.GetFirstToken();
```

Comment:
Are you causing the local declarations within the switch blocks to be in scope within the switch expression? I don't think that is the way it is supposed to be.